tests/Helpers/Factory.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace Algolia\AlgoliaSearch\Helper\Tests\Helpers;
+
+use Algolia\AlgoliaSearch\Helper\SearchClient;
+
+final class Factory
+{
+    private static $instance;
+
+    private $client;
+
+    /**
+     * Returns a new index name based on the given `name`.
+     *
+     * @param string $name
+     *
+     * @return string
+     */
+    public static function getIndexName($name)
+    {
+        if (!self::$instance) {
+            self::$instance = getenv('TRAVIS') ? getenv('TRAVIS_JOB_NUMBER') : get_current_user();
+        }
+
+        return sprintf('TRAVIS_php-helper_%s_%s_%s', date('Y-M-d_H:i:s'), self::$instance, $name);
```

Comment:
Can you validate with @aseure if this is good?

Reply:
@TomKlotzPro explained me that:

1. you are running your tests on the same AppID as the CTS 
2. you need to prefix your indices name with `TRAVIS_` otherwise the key dealer doesn't work

However, the format for the CTS is `LANG_DATE_TIME_INSTANCE_TEST` as documented here: https://github.com/algolia/algoliasearch-client-specs/tree/master/common-test-suite#index-naming-convention. If you start adding other conventions on top of it, it means we won't be able to correctly delete the indices afterwards. I'd suggest:

1. use a different appID than the CTS one for now if your tests are not passing
2. take back the ownership of the API key dealer to make it work with the conventions we introduced in the CTS
3. change back the appID to the one of the CTS

wdyt?